src/physics/cam/clubb_mf.F90

```diff
@@ -450,10 +545,76 @@ subroutine integrate_mf( nz,
        wmin = sigmaw * pwmin
        wmax = sigmaw * pwmax
 
+       do i=1,clubb_mf_nup
+         wlv = wmin + (wmax-wmin) / (real(clubb_mf_nup,r8)) * (real(i-1, r8))
+         wtv = wmin + (wmax-wmin) / (real(clubb_mf_nup,r8)) * real(i,r8)
+
+         upw(1,i) = 0.5_r8 * (wlv+wtv)
+         upa(1,i) = 0.5_r8 * erf( wtv/(sqrt(2._r8)*sigmaw) ) &
+                    - 0.5_r8 * erf( wlv/(sqrt(2._r8)*sigmaw) )
+
+         upmf(1,i)= rho_zm(1)*upa(1,i)*upw(1,i)
+
+         upu(1,i) = u(1)
+         upv(1,i) = v(1)
+
+         upqt(1,i)  = cwqt * upw(1,i) * sigmaqt/sigmaw
+         upthv(1,i) = cwthv * upw(1,i) * sigmathv/sigmaw
+       enddo
+
+       facqtu=1._r8
+       facthvu=1._r8
+       if (scalesrf) then
+         ! scale surface fluxes
+         srfwqtu = 0._r8
+         srfwthvu = 0._r8
+         srfarea = 0._r8
+         do i=1,clubb_mf_nup
+             srfwqtu=srfwqtu+upqt(1,i)*upw(1,i)*upa(1,i)

```

Comment:
This loops throughout the N updrafts but these variables assume the latest value (they don't have indexes). I don't remember what is this block for though, so ignore my comment if it doesn't make sense!